classes/manager.js

```diff
@@ -1,9 +1,25 @@
 import User from './user.js';
 
 class Manager extends User {
-    constructor() {
+  constructor(userData) {
+    super(userData)
+  }
 
-    }
+  calcPercRoomsBooked(hotel, date) {
+    const availCount = hotel.findAvailableRooms(date).length;
+    const roomCount = hotel.rooms.length;
+    return (roomCount - availCount) / roomCount;
+  }
+
+  calcTodaysRev(hotel, date) {
+    const todaysRes = hotel.findReservationsByDate(date);
+    return this.calculateRewardsTotal(todaysRes, hotel).toFixed(2);
+  }
+
+  setManagerPrivilege(user) {
+    this.id = user.id;
```

Comment:
How do you implement this method? It looks like you're setting the view from a manager to a user..?

Reply:
The manager class is inheriting from the user class, and a bunch of the user methods such as looking for all their reservations are reference the class ID.

This method takes an input user (determined by the search feature) and sets the manager ID and Name as the selected user, so that when they call the inherited user methods like find reservations and the likes, they appear as the user returning that user data. Sort of a spoofing the system.

Reply:
This is a super interesting way fo doing that, Coo!